lib/Resolver/ClassResolver.php

```diff
@@ -15,20 +14,28 @@
  *  @license    http://www.pimcore.org/license     GPLv3 and PCL
  */
 
-namespace Pimcore\Model\Document\TypeDefinition\Loader;
 
-use Pimcore\Loader\ImplementationLoader\PrefixLoader as BasePrefixLoader;
+namespace Pimcore\Resolver;
 
 /**
+ * Core class resolver returning FQCN
+ *
  * @internal
  */
-final class PrefixLoader extends BasePrefixLoader
+class ClassResolver implements ResolverInterface
 {
-    /**
-     * {@inheritdoc}
-     */
-    protected function normalizeName(string $name): string
+    public const TYPE_DOCUMENTS = 'documents';
```

Comment:
I'm just wondering whether we should consider the type in the resolver or if we should have a resolver service for each type. 

So basically having a `pimcore.class.resolver.asset` and a `pimcore.class.resolver.document` where we configure the map as a parameter (that was mapped in the extension). 

So basically, like this: 
```yaml
    pimcore.class.resolver.document:
        arguments:
            $map: '%pimcore.documents.type_definitions.map%'
```

Reply:
This would be a more generic approach I agree. The main difference is that you have to get the correct resolver for the type in the code instead of adding the type as a param. 

I am just wondering if it would make sense to have `n` (in our case most likely 2 or 3) services instead of one?

Reply:
Agreed!